client/src/pages/Property.tsx

```diff
@@ -29,17 +32,36 @@ const Property = () => {
     updatedAt: new Date(),
   });
   const [comments, setComments] = useState<Comment[]>([]);
-
+  const navigate = useNavigate();
   useEffect(() => {
-    getPostComments().then((data) => {
-      setComments(data);
-    });
+    setLoadingPost(true);
+    setError(null);
+    getPost()
+      .then((data) => {
+        setPost(data.post);
+        setComments(data.post.comments);
+        setLoadingPost(false);
+      })
+      .catch(() => {
+        setLoadingPost(false);
+        setError("An error occurred.");
+      });
   }, [comments.length]);
```

Comment:
```suggestion
  }, []);
```
This is also causing multiple API calls to `/postings` also